### PR TITLE
Moving to redux

### PR DESCRIPTION
**Pull request by Nosoyleon**
## Summary
1. Folder Structure Redefined
2. Using Redux to handle state

## Screenshots
image

### CODE REVIEW

File: react/training-project/src/app/screens/Game/components/Board/components/Square/index.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import actionCreators from '~redux/History/actions';
+
+import style from './styles.scss';
+
+class Square extends Component {
+  handleClick = () => this.props.handleClick({ ...this.props });
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/components/Board/components/Square/index.js]: Is it necessary to pass as argument all the props? :thinking:

File: react/training-project/src/app/screens/Game/components/Board/components/Square/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  xIsNext: state.xIsNext
+});
+
+const mapDispatchToProps = dispatch => ({
+  handleClick: newstate => dispatch(actionCreators.handleClick(newstate))
+});
+
+Square.propTypes = {
+  handleClick: PropTypes.func.isRequired,
+  value: PropTypes.string
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/components/Board/components/Square/index.js]: Isn't it required? If it's not and the value is `null` or `undefined`, you will be rendering a button with no text. Is it ok? :thinking:

File: react/training-project/src/app/screens/Game/index.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import Button from '~components/Button';
+
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/index.js]: Remove this empty line.

File: react/training-project/src/app/screens/Game/index.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import actionCreators from '~redux/History/actions';
+
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/index.js]: And this one too.

File: react/training-project/src/app/screens/Game/index.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+const mapStateToProps = state => ({
+  history: state.history,
+  stepNumber: state.stepNumber,
+  xIsNext: state.xIsNext
+});
+
+const mapDispatchToProps = dispatch => ({
+  jumpTo: step => dispatch(actionCreators.setJumTo(step)),
+  handleClick: newstate => dispatch(actionCreators.handleClick(newstate))
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/index.js]: typo: `newstate` -> `newState`

File: react/training-project/src/app/screens/Game/components/Board/components/Square/styles.scss [hunk 6]
```diff
@@ -1,4 +1,4 @@
-@import '../../../scss/colors';
+@import '../../../../../../../scss/_colors';
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/components/Board/components/Square/styles.scss]: Absolute import or aliases?

File: react/training-project/src/app/screens/Game/index.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          <ol>{moves}</ol>
+        </div>
+      </div>
+    );
+  }
+}
+
+Game.propTypes = {
+  history: PropTypes.arrayOf(PropTypes.object),
+  stepNumber: PropTypes.number,
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/index.js]: Isn't it required?

File: react/training-project/src/app/screens/Game/index.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  jumpTo: PropTypes.func,
+  xIsNext: PropTypes.bool
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/index.js]: And this too.

File: react/training-project/src/app/screens/Game/index.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        </div>
+      </div>
+    );
+  }
+}
+
+Game.propTypes = {
+  history: PropTypes.arrayOf(PropTypes.object),
+  stepNumber: PropTypes.number,
+  jumpTo: PropTypes.func,
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/index.js]: And this.

File: react/training-project/src/redux/History/actions.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+export const actions = {
+  HANDLE_CLICK: '@@HISTORY/HANDLE_CLICK',
+  BLOCK_ON_WINNER: '@@HISTORY/BLOCK_ON_WINNER',
+  SET_SQUARE_CLICK: '@@HISTORY/SET_SQUARE_CLICK',
+  SET_JUMP_TO: '@@HISTORY/SET_JUMP_TO'
+};
+
+const actionCreators = {
+  setSquareClick: newstate => ({
```
Comment (BJordi) [inline react/training-project/src/redux/History/actions.js]: typo: `newstate` -> `newState`

**Review by BJordi — CHANGES_REQUESTED** [review]

File: react/training-project/src/app/screens/Game/components/Board/components/Square/index.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+class Square extends Component {
+  handleClick = () =>
+    this.props.handleClick({
+      history: this.props.history,
+      id: this.props.id,
+      stepNumber: this.props.stepNumber,
+      xIsNext: this.props.xIsNext
+    });
+  render() {
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/components/Board/components/Square/index.js:17]: Add en empty line before this one.

File: react/training-project/src/app/screens/Game/components/Board/components/Square/styles.scss [hunk 12]
```diff
@@ -1,4 +1,4 @@
-@import '../../../scss/colors';
+@import 'src/scss/_colors';
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/components/Board/components/Square/styles.scss:1]: Check it, but I think you can add an env variable `NODE_PATH=src/`so you can do `scss/_colors`.

File: react/training-project/src/app/screens/Game/index.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+/* eslint-disable import/order */
```
Comment (BJordi) [inline react/training-project/src/app/screens/Game/index.js:5]: Why are you needing this? :thinking:

**Review by BJordi — APPROVED** [review]

File: react/training-project/src/app/screens/Game/index.js (hunk 13 above)
Comment (Nosoyleon) [reply react/training-project/src/app/screens/Game/index.js:5]: The linter is bothering me when I use simultaneous alias import without an empty line between.

[eslint] There should be at least one empty line between import groups [import/order]

**Pull request merged**